pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -70,8 +73,43 @@ func (r *SubnetSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		log.Error(err, "Unable to fetch SubnetSet CR", "SubnetSet", req.NamespacedName)
 		return ResultRequeue, err
 	}
+
+	hasBindings := r.subnetSetHasBindings(string(subnetsetCR.UID))
+	if len(hasBindings) > 0 {
+		if !controllerutil.ContainsFinalizer(subnetsetCR, servicecommon.SubnetSetFinalizerName) {
```

Comment:
If the finalizer exists, the SubnetSet CR is not deleted cleanly from apiserver. So no need to add the check.

Reply:
It should not happen, the reconciler logic is following this workflow
1. SubnetSet CR is deleted in apiserver (not found the CR), line 65, delete NSX resources directly.
2. SubnetSet CR still exists in apiserver:
    2.1 No finalizer exists, but binding CR exists, add finalizer. This must be not a delete SubnetSet event (otherwise it shall go into case 1)
    2.2 Finalizer already exists, binding CR exists, no need to add finalizer. This could be a SubnetSet delete or update event, but this case we don't add finalizer.
    2.3 finalizer exists, binding CRs don't exist, need to delete finalizer. This could be a SubnetSet delete or update event, so this case we can't add pre-check to only support delete case

Reply:
Same comment as the one in subnet_controller.